### PR TITLE
Add support for DNS peers

### PR DESCRIPTION
**Pull request by mattdf**
simple PR that adds domain name support to the "peer" argument, I've also made the top level function use safe_parse_peer so that the node doesn't die if one of the peers fails to resolve

### CODE REVIEW

File: apps/arweave/src/ar_util.erl [hunk 1]
```diff
@@ -124,11 +124,12 @@ parse_peer("") -> throw(empty_peer_string);
...
 parse_peer(Str) when is_list(Str) ->
-	case io_lib:fread("~d.~d.~d.~d", Str) of
-		{ok, [A, B, C, D], PortStr} ->
+    [Addr, PortStr] = parse_port_split(Str),
+    case inet:getaddr(Addr, inet) of
+		{ok, {A, B, C, D}} ->
 			{A, B, C, D, parse_port(PortStr)};
-		{error, _} ->
-			throw({invalid_peer_string, Str})
+		{error, Posix} ->
```
Comment (ldmberman) [inline apps/arweave/src/ar_util.erl:131]: Why "Posix"?

**Comment by ldmberman** [comment]
There is a failure in the build:

```
**error:{assertEqual,[{module,ar},
              {line,729},
              {expression,"Value"},
              {expected,[{ok,{5,6,7,8,9}},{ok,{1,2,3,4,1984}}]},
              {value,[{5,6,7,8,9},{1,2,3,4,1984}]}]}
  output:<<"">>
```

File: apps/arweave/src/ar_util.erl (hunk 1 above)
Comment (mattdf) [reply apps/arweave/src/ar_util.erl:131]: it's a posix return error code, i just took the name from the erlang lib description

File: apps/arweave/src/ar.erl [hunk 2]
```diff
@@ -178,7 +178,7 @@ parse_cli_args(["init"|Rest], C) ->
 parse_cli_args(["mine"|Rest], C) ->
 	parse_cli_args(Rest, C#config { mine = true });
 parse_cli_args(["peer", Peer|Rest], C = #config { peers = Ps }) ->
-	parse_cli_args(Rest, C#config { peers = [ar_util:parse_peer(Peer)|Ps] });
+	parse_cli_args(Rest, C#config { peers = [ar_util:safe_parse_peer(Peer)|Ps] });
```
Comment (vkatsuba) [inline apps/arweave/src/ar.erl:181]: `ar_util:safe_parse_peer/1` will return `{ok, Peer}` instead `Peer` - looks like by this reason you have a crash in tests

File: apps/arweave/src/ar_util.erl (hunk 1 above)
Comment (vkatsuba) [reply apps/arweave/src/ar_util.erl:131]: `posix()` is a spec for names(atoms) of POSIX error codes. More better name can be `Reason` or you can skip this `Reason` as was before

**Pull request merged**